src/06_wildfire.R

```diff
@@ -0,0 +1,569 @@
+#               destfile=use_network_path("data/Wildfires_DB/Input/Blocks_2021/ldb_000a21a_e.zip"))
+# unzip(use_network_path("data/Wildfires_DB/Input/Blocks_2021/ldb_000a21a_e.zip"))
+## READ DISSEMINATION BLOCKS
+file_path <- use_network_path("data/Wildfires_DB/Input/Blocks_2021/ldb_000b21a_e.shp")
+
+blocks <- st_read(file_path) %>%
+  filter(PRUID == '59') %>%
+  st_transform(, crs = st_crs(BC_wildfire_perimeter_historic)) %>%
+  mutate(DB_AREA = st_area(.)) 
+
+# DB_filtered <- blocks %>%
+#   select(c(DBUID, DB_AREA)) %>%
+#   st_drop_geometry()
+
+
+# Reproject datasets to a suitable CRS
+BC_wildfire_perimeter_projected <- st_transform(BC_wildfire_perimeter, crs = 32610)  
+blocks_projected <- st_transform(blocks, crs = 32610)
+
+# Ensure geometries are valid
+BC_wildfire_perimeter_projected <- st_make_valid(BC_wildfire_perimeter_projected)
+blocks_projected <- st_make_valid(blocks_projected)
+
+## COMBINE BLOCKS AND FIRES # Perform the intersection
+BC_DB_wildfire_perimeter_projected <- st_intersection(blocks_projected, BC_wildfire_perimeter_projected) %>% 
+  select(DBUID, DB_AREA,
+         FIRE_LABEL, FIRE_NUMBER, FIRE_YEAR,
+         FIRE_CAUSE, FIRE_DATE, FIRE_STATUS, TRACK_DATE, FIRE_SIZE_HECTARES,FEATURE_AREA_SQM,
+         everything()) %>% 
+  arrange(DBUID, FIRE_LABEL)
+  # select(c(DBUID, FIRE_LABEL, FIRE_NUMBER, FIRE_YEAR, FIRE_CAUSE, FIRE_DATE, FIRE_STATUS, TRACK_DATE, FIRE_SIZE_HECTARES,
+  #          FEATURE_AREA_SQM))
+
+##. Reproject back to original CRS 
+BC_DB_wildfire_perimeter <- st_transform(BC_DB_wildfire_perimeter_projected, st_crs(BC_wildfire_perimeter))
+
+
+# Get back the DB area, only areas that cover fires, one DB could have three fires in one year
+# BC_DB_wildfire_perimeter_perimeter <- DB_filtered %>% inner_join(BC_DB_wildfire_perimeter_perimeter, by = join_by(DBUID) )
+
+# ? should we group by DB to get the total percent? or if the fire areas are overlapped, should we st_union them to avoid double counting?
+# BC_DB_wildfire_perimeter_list <- split(BC_DB_wildfire_perimeter_perimeter, BC_DB_wildfire_perimeter_perimeter$FIRE_YEAR)
+
+# Group Data by Year and Region: Use dplyr::group_by() to group the spatial data by the year and region.
+# Identify Overlaps Within Groups:
+#   Use st_intersects() to identify spatial intersections between fire areas.
+# Check if any intersecting geometries have different timestamps within the same group.
+# Summarize Results: For each group, determine if overlaps exist and how many.
+
+
+# Group data by year and region, then identify overlaps
+overlap_results <- BC_DB_wildfire_perimeter %>%
+  group_by(FIRE_YEAR, DBUID) %>%
+  summarise(
+    overlaps = any(sapply(
+      seq_len(n()),
+      function(i) {
+        # Work with the current group only
+        current_group <-  st_as_sf(cur_data())  # Ensure it's an sf object # Access only rows within the group
```

Comment:
Apparently, `cur_data` is deprecated. This is equivalent:

`current_group <-  st_as_sf(pick(everything))  # Ensure it's an sf object # Access only rows within the group`

Reply:
Oh. It works. I have not picked up 'pick' yet.